spec/form_models/application_details_form_spec.rb

```diff
@@ -43,20 +135,41 @@
     describe '#publish_on' do
       let(:application_details) { ApplicationDetailsForm.new(publish_on: Time.zone.yesterday) }
 
-      it 'the publish date date must be present' do
+      it 'the publish date must be present' do
         expect(application_details.valid?).to be false
         expect(application_details.errors.messages[:publish_on][0])
           .to eq('can\'t be before today')
       end
     end
   end
 
+  context 'when expiry time components are given' do
+    it 'cannot save expiry time if fields are incomplete' do
+      application_details = ApplicationDetailsForm.new(contact_email: 'some@email.com', expiry_time_hh: '9')
+
+      expect(application_details.params_to_save).to eq(contact_email: 'some@email.com')
```

Comment:
Is `application_details.params_to_save` expected to return an array? 
Could we use the `.not_to include` RSpec matcher here?

Reply:
Thanks, reads much better, resolved this in e543ec7